snakemake/io.py

```diff
@@ -1247,12 +1247,18 @@ def path_to_str(f):
 
     filepatterns = list(map(path_to_str, filepatterns))
 
-    if any(map(lambda f: getattr(f, "flags", {}), filepatterns)):
-        raise WorkflowError(
-            "Flags in file patterns given to expand() are invalid. "
-            "Flags (e.g. temp(), directory()) have to be applied outside "
-            "of expand (e.g. 'temp(expand(\"plots/{sample}.pdf\", sample=SAMPLES))')."
-        )
+    for filepattern in filepatterns:
+        filepattern_flags = {
+            key: value
+            for key, value in getattr(filepattern, "flags", {}).items()
+            if key not in ["path_modified"]
```

Comment:
Maybe rather this?
```suggestion
            if key != PATH_MODIFIER_FLAG
```

Reply:
Was not sure if there might be other flags that should be ignored, but it can be changed later if needed.